app/views.py

```diff
@@ -0,0 +1,204 @@
+            return make_response(jsonify(response)), 200
+
+        elif request.method == "GET":
+            # Fetch the specified bucketlist
+            related_items = BucketListItem.query.filter_by(
+                bucketlist_id=id)
+            results = []
+            for item in related_items:
+                details = {
+                    "id": item.id,
+                    "name": item.name,
+                    "date_created": item.date_created,
+                    "date_modified": item.date_modified,
+                    "done": item.done
+                }
+                results.append(details)
+            response = {
+                "id": bucketlist.id,
+                "name": bucketlist.name,
+                "items": results,
+                "date_created": bucketlist.date_created,
+                "date_modified": bucketlist.date_modified,
+                "created_by": bucketlist.created_by
+            }
+            return make_response(jsonify(response)), 200
+
+        else:
+            # delete the bucketlist using our delete method
+            bucketlist.delete()
+            return {"message": "bucketlist {} deleted".
+                    format(bucketlist.id)}, 200
+
+    @app.route("/bucketlists/<int:id>/items/", methods=["POST"])
+    @decorator.auth_token
+    def bucketlist_item(id, **kwargs):
+        user_id = kwargs["user_id"]
+        bucketlist = BucketList.query.filter_by(
+            id=id, created_by=user_id).first()
+
+        if not bucketlist:
+            # Raise an HTTPException with a 404 not found status code
+            abort(404)
+
+        else:
+            name = str(request.data.get("name", ""))
+            if name:
+                item = BucketListItem(name=name, bucketlist_id=id)
+                item.save()
+                response = jsonify({
+                    "id": item.id,
+                    "name": item.name,
+                    "date_created": item.date_created,
+                    "date_modified": item.date_modified,
+                    "bucketlist_id": id,
+                    "done": item.done
+                })
+                return make_response(response), 201
+
+    @app.route("/bucketlists/<int:id>/items/<int:item_id>",
+               methods=["PUT", "GET", "DELETE"])
+    @decorator.auth_token
+    def bucketlist_item_manipulation(id, item_id, **kwargs):
+        user_id = kwargs["user_id"]
+        bucketlist = BucketList.query.filter_by(
+            id=id, created_by=user_id).first()
+        if not bucketlist:
+            # Raise an HTTPException with a 404 not found status code
+            abort(404)
+        else:
+            item = BucketListItem.query.filter_by(id=item_id).first()
+            if not item:
+                # Raise an HTTPException
+                abort(404)
+
+            if request.method == "PUT":
+                # Update bucketlist with new name
+                name = str(request.data.get("name", ""))
```

Comment:
I thinks this would update the name even if the user doesn't enter anything. also check that all post data is validated.